### DISCUSSION TITLE
Multiple remote_addrs and local_addrs in tunnel mode?

### DISCUSSION THREAD

**Discussion by andy-ning** [strongswan/strongswan#2515] [General]
To secure traffic between two sites in tunnel mode, site A has two IP addresses and site B has one IP address (there are traffics on both IP addresses on site A). Is it possible for local_addrs (on site A), and remote_addrs (on site B) to have multiple addresses? Or there should be two separate connections configured?

For a setup like this, what would be a valid configuration?
`10.1.0.0/16 -- | 192.168.0.1, 192.168.0.2 | === | 192.168.0.3 | -- 10.2.0.0/16`
`  moon-net         moon                                sun              sun-net  `

**Comment by tobiasbrunner** [comment]
> (there are traffics on both IP addresses on site A)

What does that mean in regards to IKE/IPsec?

> Is it possible for local_addrs (on site A), and remote_addrs (on site B) to have multiple addresses?

To what end?

**Reply by andy-ning** [reply]
> What does that mean in regards to IKE/IPsec?

There are traffics between moon's two IPs and sun's one IP. (in the example, there are traffics between 192.168.0.1 and 192.168.0.3, and traffics between 192.168.0.2 and 192.168.0.3). We need to secure all these traffics using **IPsec tunnel mode**. So do we need to define two connections, one for each pair, or one connection can cover both traffics?

For example, will the following configuration work?
`Moon:
  connections {
    net-net {
      remote_addrs = 192.168.0.3
      **local_addrs = 192.168.0.1, 192.168.0.2**
      local {
        auth = pubkey
        certs = moonCert.pem
      }
      remote {
        auth = pubkey
        id = "C=CH, O=strongSwan, CN=sun.strongswan.org"
      }
      children {
        net-net {
          local_ts  = 10.1.0.0/16
          remote_ts = 10.2.0.0/16
          start_action = trap
        }
      }
    }
  }

Sun:
  connections {
    net-net {
      **remote_addrs = 192.168.0.1, 192.168.0.2**
      local_addrs = 192.168.0.3
      local {
        auth = pubkey
        certs = sunCert.pem
      }
      remote {
        auth = pubkey
        id = "C=CH, O=strongSwan, CN=moon.strongswan.org"
      }
      children {
        net-net {
          local_ts  = 10.2.0.0/16
          remote_ts = 10.1.0.0/16
          start_action = trap
        }
      }
    }
  }`

**Reply by tobiasbrunner** [reply]
To protect additional traffic, you add those subnets/IPs to the traffic selectors (i.e. `local|remote_ts`), not `local|remote_addrs`, which defines the endpoints of the VPN connection, not what traffic is protected.

**Reply by andy-ning** [reply]
That makes sense. But is my example configuration valid? (with multiple IPs in local|remote_addrs for tunnel mode)

**Reply by tobiasbrunner** [reply]
It's a valid config, but it won't change anything (only the first address is generally used, except for `remote_addrs` as responder when searching configs for connection requests).
